frontend/packages/dev-console/src/components/import/builder/BuilderImageEnvironments.tsx

```diff
@@ -0,0 +1,58 @@
+import * as React from 'react';
+import { TextInputTypes } from '@patternfly/react-core';
+import { useTranslation } from 'react-i18next';
+import {
+  ImportEnvironment,
+  isImportEnvironment,
+  useResolvedExtensions,
+} from '@console/dynamic-plugin-sdk';
+import { InputField } from '@console/shared';
+
+interface BuilderImageEnvironmentsProps {
+  name: string;
+  imageStreamName: string;
+  imageStreamTag: string;
+}
+
+const BuilderImageEnvironments: React.FC<BuilderImageEnvironmentsProps> = ({
+  name,
+  imageStreamName,
+  imageStreamTag,
+}) => {
+  const { t } = useTranslation();
+  const [environmentExtensions, resolved] = useResolvedExtensions<ImportEnvironment>(
+    isImportEnvironment,
+  );
+
+  const filteredExtensions = React.useMemo(
+    () =>
+      environmentExtensions?.filter(
+        (e) =>
+          e.properties.imageStreamName === imageStreamName &&
+          e.properties.imageStreamTags.includes(imageStreamTag),
+      ),
+    [environmentExtensions, imageStreamName, imageStreamTag],
+  );
+
+  if (!resolved) {
+    return null;
+  }
+  return (
+    <>
+      {filteredExtensions.map(({ properties }) =>
+        properties.environments.map((env) => (
+          <InputField
+            key={`${properties.imageStreamName}-${env.key}`}
+            type={TextInputTypes.text}
+            name={`${name}.${env.key}`}
+            label={t(env.label)}
+            helpText={env.description ? t(env.description) : null}
```

Comment:
I don't think translate is needed here for env.description it's done when extension is defined

Reply:
Updated.